Jenkinsfile

```diff
@@ -0,0 +1,223 @@
+      keepLongStdio: true,
+      testDataPublishers: [[$class: 'StabilityTestDataPublisher']],
+      testResults: "**/${surefireTestReports}"
+}
+
+if (currentBuild.result.equals('FAILURE')){
+  return 1
+}
+
+try {
+  timestamps {
+    node {
+      ansicolor {
+
+        stage('Integration tests') {
+          def tasks = [:]
+          tasks['Integration tests: WILDFLY'] = {
+            node {
+              ansicolor {
+                info.printNode()
+                info.printEnv()
+                debugChromeDriver()
+                unstash 'workspace'
+                integrationTests('wildfly8')
+              }
+            }
+          }
+          tasks['Integration tests: JBOSSEAP'] = {
+            node {
+              ansicolor {
+                info.printNode()
+                info.printEnv()
+                debugChromeDriver()
+                unstash 'workspace'
+                integrationTests('jbosseap6')
+              }
+            }
+          }
+          tasks.failFast = true
+          parallel tasks
+        }
+      }
+      // TODO in case of failure, notify culprits via IRC and/or email
+      // https://wiki.jenkins-ci.org/display/JENKINS/Email-ext+plugin#Email-extplugin-PipelineExamples
+      // http://stackoverflow.com/a/39535424/14379
+      // IRC: https://issues.jenkins-ci.org/browse/JENKINS-33922
+      // possible alternatives: Slack, HipChat, RocketChat, Telegram?
+      notify.successful()
+    }
+  }
+} catch (e) {
+  notify.failed()
+  throw e
+}
+
+// TODO factor these out into zanata-pipeline-library too
+
+void xvfb(Closure wrapped) {
+  wrap([$class: 'Xvfb']) {
+    wrapped.call()
+  }
+}
+
+void debugChromeDriver() {
+  sh returnStatus: true, script: 'which chromedriver google-chrome'
+  sh returnStatus: true, script: 'ls -l /opt/chromedriver /opt/google/chrome/google-chrome'
+}
+
+void integrationTests(def appserver) {
+  sh "find . -path \"*/${failsafeTestReports}\" -delete"
+
+  try{
+    xvfb {
+      withPorts {
+        // Run the maven build
+        sh """./run-clean.sh ./mvnw -e verify \
+                   --batch-mode \
+                   --settings .travis-settings.xml \
+                   -Danimal.sniffer.skip=true \
+                   -DstaticAnalysis=false \
+                   -Dcheckstyle.skip \
+                   -Dappserver=$appserver \
+                   -Dcargo.debug.jvm.args= \
+                   -DskipUnitTests \
+                   -Dmaven.main.skip \
+                   -Dgwt.compiler.skip \
+                   -Dwebdriver.display=${env.DISPLAY} \
+                   -Dwebdriver.type=chrome \
+                   -Dwebdriver.chrome.driver=/opt/chromedriver \
+                   -DallFuncTests
+          """
+      }
+    }
+  }catch(e){
+    currentBuild.result = 'UNSTABLE'
+    archiveTestFilesIfUnstable()
+    throw e
+  }finally{
+    notify.testResults(appserver.toUpperCase())
+    junit allowEmptyResults: true,
+        keepLongStdio: true,
```

Comment:
http://javadoc.jenkins-ci.org/junit/hudson/tasks/junit/JUnitParser.html#JUnitParser-boolean-boolean-

Having looked into it, I don't think we need the option keepLongStdio, for failsafe, or for surefire above. Could you remove them please @definite ?